src/instructlab/model/chat.py

```diff
@@ -61,6 +62,30 @@
 
 PROMPT_PREFIX = ">>> "
 
+DEFAULT_ENDPOINT = "http://127.0.0.1:8000/v1"
+
+
+def is_openai_server_and_serving_model(
+    endpoint: str, api_key: str, http_params: HttpClientParams
+) -> bool:
+    """
+    Given an endpoint, returns whether or not the given response is expected
+    from an OpenAI-compatible server, and whether or not it's actively serving
+    a model.
+    """
+    client = OpenAI(
+        base_url=endpoint,
+        api_key=api_key,
+        timeout=cfg.DEFAULTS.CONNECTION_TIMEOUT,
+        http_client=http_client(http_params),
+    )
+    try:
+        # ensure the model specified exists on the server. with backends like vllm, this is crucial.
+        model_list = client.models.list().data
+        return len(model_list) > 0
```

Comment:
nit: `return bool(model_list)` since it doesn't seem important which length it has as long as it's not empty.

Reply:
While that would functionally work, we should keep it as is since it's descriptive about what the exact condition is.